src/lyra/abstract_domains/liveness/liveness_domain.py

```diff
@@ -209,14 +209,26 @@ def _substitute(self, left: Expression, right: Expression) -> 'StrongLivenessSta
             return self
         elif isinstance(left, Subscription) or isinstance(left, Slicing):
             target = left.target
-            if self.store[target].is_top():  # the assigned variable is strongly-live
-                # summarization abstraction
+            if self.store[target].is_top():  # the target variable (list, dict,..) is strongly-live
+                # summarization abstraction (weak update)
                 for identifier in right.ids():
                     if isinstance(identifier, VariableIdentifier):
                         self.store[identifier].top()
                     else:
                         error = f"Substitution with {right} is not implemented!"
                         raise NotImplementedError(error)
+
+                if (isinstance(left, Subscription)):
+                    ids = left.key.ids()
+                else:  # Slicing
+                    ids = left.lower.ids() | left.upper.ids()
+                for identifier in ids:  # make ids in subscript strongly live
+                    # needed? ids = VariableIdentifier?
```

Comment:
ids() only collects variable identifiers so you do not need the case distinction below